lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -107,16 +107,17 @@ static SkAlphaType ChooseCompatibleAlphaType(SkAlphaType type) {
   return type;
 }
 
-std::optional<DecompressResult> ImageDecoderImpeller::DecompressTexture(
+DecompressResult ImageDecoderImpeller::DecompressTexture(
     ImageDescriptor* descriptor,
     SkISize target_size,
     impeller::ISize max_texture_size,
     bool supports_wide_gamut,
     const std::shared_ptr<impeller::Allocator>& allocator) {
   TRACE_EVENT0("impeller", __FUNCTION__);
   if (!descriptor) {
-    FML_DLOG(ERROR) << "Invalid descriptor.";
-    return std::nullopt;
+    std::string decode_error("Invalid descriptor");
```

Comment:
This is a "this should never happen and is a bug in the engine code." I'm not sure if we need to say that here or not.